app/models/framework/management_charge_calculator/column_based.rb

```diff
@@ -0,0 +1,34 @@
+class Framework
+  module ManagementChargeCalculator
+    class ColumnBased
+      attr_reader :column, :value_to_percentage
+
+      def initialize(column:, value_to_percentage:)
+        @column = column
+        @value_to_percentage = prepare_hash(value_to_percentage)
```

Comment:
Ah, I like this solution! Tempted to refactor the dependent field inclusion validator now to use a similar method. :D